### PR TITLE
[iOS] Add support for displaying `brave://flags` from the omnibox

### PR DESCRIPTION
**Pull request by kylehickinson**
This change also adds the ability to open `brave://histograms`, `brave://local-state` and `brave://version` from the omnibox.

Resolves https://github.com/brave/brave-browser/issues/37149

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by Brandon-T — APPROVED** [review]

File: ios/brave-ios/Sources/Brave/Frontend/Browser/BrowserViewController/BVC+ToolbarDelegate.swift [hunk 1]
```diff
@@ -259,45 +259,92 @@ extension BrowserViewController: TopToolbarDelegate {
...
+    return true
+  }
+
+  /// Handles displaying a Chromium web view for brave:// url that would display WebUI
+  func handleChromiumWebUIURL(_ url: URL) -> Bool {
+    let supportedPages = [
+      "flags",
+      "histograms",
+      "local-state",
+      "version",
```
Comment (Brandon-T) [inline ios/brave-ios/Sources/Brave/Frontend/Browser/BrowserViewController/BVC+ToolbarDelegate.swift:315]: Most of the URLs from `brave://chrome-urls/` will also work :)
Not sure if we need them though, but if we do, we can just add them here

Comment (kylehickinson) [reply ios/brave-ios/Sources/Brave/Frontend/Browser/BrowserViewController/BVC+ToolbarDelegate.swift:315]: Yeah technically we don't _need_ this whitelist at all if we can have `brave://sync-internals` load like normal

**Review by goodov — APPROVED** [review]
I don't like that we had to duplicate already declared flags from the desktop/android impl, but I guess it's the Chromium way..

File: ios/brave-ios/Sources/Brave/Frontend/Browser/BrowserViewController/BVC+ToolbarDelegate.swift (hunk 1 above)
Comment (stoletheminerals) [reply ios/brave-ios/Sources/Brave/Frontend/Browser/BrowserViewController/BVC+ToolbarDelegate.swift:315]: The question is do we need other brave:// URLs available on iOS? I'd keep it limited if we don't need them. I'll need to look at sync-internals more closely, there is some data exporting functionalities that we may need to hide behind biometrics on Android.

**Comment by kjozwiak** [comment]
Verification **PASSED** on `iPhone 11` running `iOS 17.4.1` using the following build(s):

```
1.66.62 Chromium: 123.0.6312.105 (Official Build) nightly (64-bit)
--- | ---
Revision | e8a70be9c002...
OS | iOS
```

* ensured that you can load `brave://skus-internals` without issues
* ensured that you can load `brave://version` without issues
* ensured that you can load `brave://local-state` without issues
* ensured that you can load `brave://histograms` without issues
* ensured that you can load `brave://flags` without issues

**`Example`** | **`Example`** | **`Example`** | **`Example`**
---------------|-----------------|----------------|-----------------
IMG_0233 | IMG_0234 | IMG_0235 | IMG_0236 | IMG_0237

**Pull request merged**